common/presenters/table/object-to-table-head.js

```diff
@@ -1,17 +1,61 @@
 const { pickBy } = require('lodash')
+const queryString = require('query-string')
 
 const i18n = require('../../../config/i18n')
 
-function objectToTableHead({ head } = {}) {
-  if (!head) {
-    return undefined
+function buildUrl({ key }, query = {}) {
+  return queryString.stringify({
+    ...query,
+    sortBy: key,
+    sortDirection: query.sortDirection === 'asc' ? 'desc' : 'asc',
+  })
+}
+
+function buildAriaLabel(head, query) {
+  // the label indicates the result of the sorting action triggered by the link,
+  // which is the opposite of the current state
+  return `${i18n.t('sort')} ${i18n.t(head.html).toLowerCase()} ${
+    query.sortDirection === 'asc' ? i18n.t('descending') : i18n.t('ascending')
+  }`
+}
+
+function buildWrapper(head, query) {
+  const label = i18n.t(head.html)
+  if (!head.isSortable) {
+    return label
+  }
+  const url = buildUrl(head, query)
+  const ariaLabel = buildAriaLabel(head, query)
+  return `<a aria-label="${ariaLabel}" class="sortable-table__button" role="button" href="?${url}">${label}</a>`
+}
+
+function assignSortAttributes(head, query) {
+  if (!head.isSortable) {
+    return head.attributes ? { ...head.attributes } : undefined
```

Comment:
I wonder if this could be neatened up using defaults. There's a lot of ternaries within this file now, and as Alex said, I'm similar in that I find them harder to read when there are lots of them.

Reply:
You could destructure the argument and possible use a separate variable for the second ternary to make it a bit easier to read I think:

```javascript
function assignSortAttributes({ attributes = {}, key, isSortable }, { sortBy, sortDirection }) {
  if(!isSortable) {
    return attributes
  }

  const isCurrentSortColumn = key === sortBy
  const ariaSortDirection = sortDirection === 'asc' ? 'ascending' : 'descending'

  return {
    ...attributes,
    'aria-sort': isCurrentSortColumn 
      ? ariaSortDirection
      : 'none',
  }
}
```